gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala

```diff
@@ -0,0 +1,201 @@
+            } match {
+              case Some(blob) =>
+                HeadResult.Exists(blob.getSize)
+              case None =>
+                HeadResult.NotFound
+            }
+          }
+        }
+      }
+      .handleError {
+        case e: GcStorageException =>
+          logger.error(s"Error while checking presence of file $sha256 in GCS", e)
+          Either.left[StorageException, HeadResult] {
+            StorageException.InvalidResponseException(e.getCode, e.getMessage, e.getReason)
+          }
+      }
+  }
+
+  override def get(sha256: Sha256, dest: File): F[Either[StorageException, GetResult]] = {
+    logger.debug(s"Downloading file $sha256 from GCS")
+
+    val objectPath: String = composeObjectPath(sha256)
+
+    {
+      Option {
+        bucket.get(objectPath)
+      } match {
+        case Some(blob) =>
+          receiveStreamedFile(blob, dest, sha256)
+        case None =>
+          F.pure[Either[StorageException, GetResult]](Right(GetResult.NotFound))
+      }
+    }.handleError {
+      case e: GcStorageException =>
+        logger.error(s"Error while downloading file $sha256 from GCS", e)
+        Either.left[StorageException, GetResult] {
+          StorageException.InvalidResponseException(e.getCode, e.getMessage, e.getReason)
+        }
+    }
+  }
+
+  private def receiveStreamedFile(blob: Blob, destination: File, expectedHash: Sha256): F[Either[StorageException, GetResult]] = {
+    logger.debug(s"Downloading streamed data to $destination")
+
+    val openOptions = Seq(StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
+
+    blocker
+      .blockOn {
+        F.delay(destination.newOutputStream(openOptions))
+          .bracket { fileStream =>
+            F.delay(new DigestOutputStream(fileStream, MessageDigest.getInstance("SHA-256")))
+              .bracket { stream =>
+                F.delay(blob.downloadTo(stream)).map { _ =>
+                  (blob.getSize, Sha256(stream.getMessageDigest.digest))
+                }
+              }(stream => F.delay(stream.close()))
+          }(fileStream => F.delay(fileStream.close()))
+      }
+      .map {
+        case (size, hash) =>
+          if (expectedHash != hash) {
+            Left {
+              StorageException.InvalidDataException(200, "-stream-", s"Expected SHA256 $expectedHash but got $hash")
+            }
+          } else {
+            Right {
+              GetResult.Downloaded(destination, size)
+            }
+          }
+      }
+  }
+
+  override def close(): Unit = {
+    ()
+  }
+}
+
+object GcsStorageBackend {
+  def fromConfig[F[_]: ConcurrentEffect: ContextShift](
+      config: Config,
+      blocker: Blocker): Either[ConfigurationException, Resource[F, GcsStorageBackend[F]]] = {
+
+    def composeConfig: Either[ConfigurationException, GcsBackendConfiguration] = {
+      val DefaultConfig = ConfigFactory.defaultReference().getConfig("gcsBackendDefaults")
```

Comment:
this belongs to private final val inside of the object

Reply:
Changed, but the original code was copy-pasted from your HCP backend implementation 😉